libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/RouteRefreshExtra.kt

```diff
@@ -0,0 +1,54 @@
+package com.mapbox.navigation.core.routerefresh
+
+import androidx.annotation.StringDef
+import com.mapbox.api.directions.v5.models.Incident
+import com.mapbox.api.directions.v5.models.LegAnnotation
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.route.RouterOrigin
+import com.mapbox.navigation.core.MapboxNavigation
+
+/**
+ * Extra data of route refresh
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+object RouteRefreshExtra {
+
+    /**
+     * Whenever one of the following conditions is `true` route refresh controller is in idle:
+     * - Trip session is not started. See [MapboxNavigation.startTripSession];
+     * - [MapboxNavigation.setNavigationRoutes] is not invoked or a list or the routes is empty;
+     * - [RouteOptions.enableRefresh] is not `true`;
+     * - Route origin of the route is [RouterOrigin.Onboard], so it cannot be refreshed from
+     * Directions API;
+     * - Refresh controller becomes idle between stages when routes are refreshed and a new round or
```

Comment:
Shouldn't it be `REFRESH_STATE_UP_TO_DATE ` state?
```suggestion
```

Reply:
I'm asking this because I imagined following application needs:
* be aware the route refresh doesn't work: Idle state
* be aware that route refresh works and route is fresh
* be aware that route refresh works but route isn't fresh 

Which application scenarios did you consider?

Reply:
I think it should, but now it's hard to handle as route refresh is actually stopped when routes are refreshed/failed. We must address it along with https://github.com/mapbox/mapbox-navigation-android/pull/6345#discussion_r974007115

Reply:
Is it really stopped? The test `after invalidation route isn't updated until successful refresh` says that it doesn't stop after failure 🤔

Reply:
I see what you mean, on successful refresh, function returns and wait for the route observer to be triggered to refresh the next route

Reply:
Do you think it's a blocker to implement keep state "up to date" between route refreshes?

Reply:
as we cannot fully control route refresh controller (like `start`-`stop`) I would avoid try to find tricky ways to know if we're still refreshing or already stopped and spread real state. It's actually easy to fix if we implement https://github.com/mapbox/mapbox-navigation-android/pull/6345#discussion_r974007115

Reply:
why we cannot fully control route refresh? this process happens only inside SDK, doesn't it?

Reply:
>Which application scenarios did you consider?

acciring to the customer needs they need to know when route is refreshed or annotations/incidents are gonna be cleaned up. So `idle` state is not so important